lib/config_generator.py

```diff
@@ -0,0 +1,105 @@
+import re
+import os
+from pathlib import Path
+
+ADDONS_DIR = os.path.abspath('../..')
+
+ACF_GUNS_PATHS = ADDONS_DIR + "/acf/lua/acf/shared/guns"
+STUBS_PATH = ADDONS_DIR + "/cfc_acf_stubber/lua/cfc_acf_stubber/stubs"
+
+PATTERN_DEFINE_GUN_CLASS = "ACF_defineGunClass\\(\"([\\w\\.]+)\", {\\s*\n([^\\)]+)} \\)"
+PATTERN_DEFINE_GUN = "ACF_defineGun\\(\"([\\w\\.]+)\", {\\s*\n([^\\)]+)} \\)"
+
+PATTERN_KEY_VALUES = "(?:\\s+([\\w]+)\\s?=\\s?(.+))|(})"
+
+PATTERN_COMMENT_BLOCK = "--\\[\\[[^\\]]+\\]\\](?:--)?"
+PATTERN_COMMENT_SINGLE = "--[^\n]+"
```

Comment:
Maybe a `PATTERNS` struct would be good here?

Reply:
Solid call, will do